### PR TITLE
 ICMSLST-887 Add postcode lookup to Manufacturer details section

### PR DESCRIPTION
**Pull request by MattHolmes123**
Changes:
  - Refactor python postcode to address view to be more robust.
  - Add vanilla js api to fetch postcode data.
  - Add postcode lookup to manufacturer details section of CFS application.
  - Refactor user edit page to use new postcode lookup.
  - Refactor User views (views at the top, helper function below)

### CODE REVIEW

File: web/static/web/js/pages/cfs-manufacturer-address.js [hunk 1]
```diff
@@ -10,17 +16,105 @@ window.addEventListener('load', (event) => {
...
+          clearErrors();
+          if (response.ok) {
+            setAddressSelectDisplay("block")
+            processAddressOptions(response);
+          } else {
+            setAddressSelectDisplay("none")
+            processError(response);
+          }
+      });
     }, 1000);
```
Comment (SamDudley) [inline web/static/web/js/pages/cfs-manufacturer-address.js:49]: What about 750?

File: web/static/web/js/pages/cfs-manufacturer-address.js [hunk 2]
```diff
@@ -10,17 +16,105 @@ window.addEventListener('load', (event) => {
...
+
+
+function processAddressOptions(response) {
+  const choicesSelect = document.querySelector("#id_available_addresses");
+
+  // reset the choices if present
+  choicesSelect.length = 0
+
+  // Iterate over the results to create the available choices
+  response.json().then((results) => {
```
Comment (SamDudley) [inline web/static/web/js/pages/cfs-manufacturer-address.js:74]: Move `.json()` call up a level.

File: web/templates/partial/js_constants.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<script type="application/javascript">
+  const API_ENDPOINTS = Object.freeze({
```
Comment (SamDudley) [inline web/templates/partial/js_constants.html:2]: Add comment above that this object should contain `key, url`.

**Review by SamDudley — CHANGES_REQUESTED** [review]

**Review by SamDudley — APPROVED** [review]
Looks great to me!

**Review by oskusalerma — APPROVED** [review]

**Review by kevinetienne — APPROVED** [review]

**Pull request merged**